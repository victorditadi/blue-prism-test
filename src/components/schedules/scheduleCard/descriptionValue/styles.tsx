import styled from 'styled-components';

export const ScheduleValue = styled.span`
  font-size: ${(props) => props.theme.size.small};
  font-family: ${(props) => props.theme.font};
`;

export const TooltipContainer = styled.div`
  visibility: hidden;
  display: none;
`;

export const Container = styled.div`
  ${ScheduleValue}:hover + ${TooltipContainer} {
    visibility: visible;
    display: flex;
    position: absolute;
    font-family: ${(props) => props.theme.font};
    font-size: ${(props) => props.theme.size.small};
    background-color: ${(props) => props.theme.colors.cyan};
    width: 15vw;
    padding: 10px 10px;
    border-radius: 4px;
    border: 1px solid ${(props) => props.theme.colors.cyanDarker};
  }
`;
