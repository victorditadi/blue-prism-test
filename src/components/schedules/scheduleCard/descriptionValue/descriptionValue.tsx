import React from 'react';

import { Container, TooltipContainer, ScheduleValue } from './styles';

const DescriptionValue = ({ text, ...props }: { text: string }) => {
  const tooltipOn = text.length > 25;
  const shortText = tooltipOn ? `${text.slice(0, 25)}...` : text;

  return (
    <Container {...props}>
      <ScheduleValue data-testid="short-text">{shortText}</ScheduleValue>
      {tooltipOn && <TooltipContainer data-testid="tooltip">{text}</TooltipContainer>}
    </Container>
  );
};

export { DescriptionValue };
