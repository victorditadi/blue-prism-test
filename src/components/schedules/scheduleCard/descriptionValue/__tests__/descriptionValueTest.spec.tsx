import { fireEvent, render } from '@testing-library/react';

import Theme from 'css/theme';
import { DescriptionValue } from '../descriptionValue';

describe('Component: DescriptionValue', () => {
  const renderComponent = ({ text }: { text: string }) =>
    render(
      <Theme>
        <DescriptionValue text={text} />
      </Theme>
    );

  it('should render DescriptionValue component', () => {
    const { asFragment } = renderComponent({ text: 'small text' });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should contain Text', () => {
    const { queryByTestId } = renderComponent({ text: 'small text' });
    expect(queryByTestId('short-text')).toBeTruthy();
  });

  it('should slice Text if Lenght > 25', () => {
    const { queryByTestId } = renderComponent({
      text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    });

    expect(queryByTestId('short-text')?.textContent).toStrictEqual('Lorem ipsum dolor sit ame...');
  });

  it('should not trigger tooltip outside mouseover', async () => {
    const { queryByTestId } = renderComponent({
      text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    });

    const tooltipContainer = queryByTestId('tooltip');
    expect(tooltipContainer).not.toBeVisible();
  });

  it('should trigger tooltip when mouseover in a long text', async () => {
    const { queryByTestId } = renderComponent({
      text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
    });
    const shortText = queryByTestId('short-text');
    shortText && fireEvent.mouseOver(shortText);

    const tooltipContainer = queryByTestId('tooltip');
    expect(tooltipContainer).toBeVisible;
  });
});
