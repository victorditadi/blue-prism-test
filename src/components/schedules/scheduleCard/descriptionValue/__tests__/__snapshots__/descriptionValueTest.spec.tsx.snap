// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DescriptionValue should render DescriptionValue component 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1em;
  font-family: Robotosans-serif;
}

<div
    class=""
  >
    <span
      class="c0"
      data-testid="short-text"
    >
      small text
    </span>
  </div>
</DocumentFragment>
`;
